components/compositing/compositor.rs

```diff
@@ -356,9 +356,7 @@ impl<Window: WindowMethods> IOCompositor<Window> {
 
     fn start_shutting_down(&mut self) {
         debug!("Compositor sending Exit message to Constellation");
-        if let Err(e) = self.constellation_chan.send(ConstellationMsg::Exit) {
-            warn!("Sending exit message to constellation failed ({}).", e);
-        }
+        self.constellation_chan.send(ConstellationMsg::Exit);
```

Comment:
My concern with switching to unmodified crossbeam channels everywhere is that we lose quite a bit of useful introspection into the current state of the system when things go wrong. @stjepang would it be possible to add an API to crossbeam-channel that automatically includes the cross-thread dropping checks and meaningful send results so that we don't have to keep reinventing it?